XMonad/Util/XUtils.hs

```diff
@@ -241,8 +241,8 @@ showSimpleWindow WindowConfig{..} strs = do
 
 -- | Like 'showSimpleWindow', but fully manage the window; i.e., destroy
 -- it after the given function finishes its execution.
-withSimpleWindow :: WindowConfig -> [String] -> X a -> X a
-withSimpleWindow wc strs doStuff = do
+withSimpleWindow :: WindowConfig -> [String] -> X a -> X (Maybe a)
+withSimpleWindow wc strs doStuff = userCode $ do
```

Comment:
Not sure if this is useful at all. A bracket would ensure the window is destroyed when `doStuff` throws, but that won't happen here, so it probably does nothing useful? I'd either move the `userCode` to just wrap `doStuff` (and then be a bit uncomfortable about the fact that there's still other stuff between create/destroy like showWindow/paintWindow which isn't surrounded by catch, but code like this worked before so maybe not a problem really) or just drop it. :-)

Reply:
I think you can only reach code like this via a handler or hook that already goes through `userCode`.

Reply:
> I think you can only reach code like this via a handler or hook that already goes through `userCode`.

Indeed, but you may still want to minimize the damage if you expect something to actually fail in a real-world usage. (Which I'm not really sure can happen here, hence my suggestion to just drop it and forget it, until perhaps we get a report of it happening.)

Reply:
Mh I guess you're right; I'll remove it and if this blows up we'll have an excuse to release 0.17.2!